File: backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any

class MessageCreate(BaseModel):
    prompt: str
    conversation_id: Optional[str] = None

class MessageResponse(BaseModel):
    id: str
    sender: str # 'user' or 'ai'
    text: str
    timestamp: str
    table_data: Optional[Dict[str, Any]] = None
    formula_code: Optional[str] = None
    follow_up_prompts: Optional[List[str]] = None

class ConversationResponse(BaseModel):
    id: str
    title: str
    category: str
    pinned: bool
    last_message: str
    timestamp: str
    messages: List[MessageResponse]

File: frontend/src/types/ai.ts
export interface AIMessage {
  id: string;
  sender: 'user' | 'ai';
  text: string;
  timestamp: string;
  bookmarked?: boolean;
  tableData?: {
    headers: string[];
    rows: string[][];
  };
  formulaCode?: string;
  followUpPrompts?: string[];
}

export interface AIConversation {
  id: string;
  title: string;
  lastMessage: string;
  timestamp: string;
  pinned: boolean;
  category: 'Budgeting' | 'Investing' | 'Taxes' | 'General' | 'Quiz Preparation';
  messages: AIMessage[];
}

export interface AIPromptItem {
  id: string;
  title: string;
  description: string;
  category: 'Budgeting' | 'Investing' | 'Taxes' | 'Calculations' | 'Career & Income' | 'FIRE Movement';
  promptText: string;
  popularCount: number;
}

export interface AIRecommendation {
  id: string;
  type: 'Lesson' | 'Book' | 'Quiz' | 'Path' | 'Assessment' | 'Action';
  title: string;
  description: string;
  reason: string;
  targetUrl: string;
  gradient: string;
}

export interface AIInsight {
  id: string;
  title: string;
  description: string;
  category: 'Improvement' | 'Weakness' | 'Strength' | 'Streak';
  metric: string;
  date: string;
}

export interface AISettingsConfig {
  responseLength: 'Concise' | 'Detailed' | 'Comprehensive';
  learningStyle: 'Practical & Examples' | 'Theoretical & Equations' | 'Balanced';
  tone: 'Encouraging & Friendly' | 'Professional & Direct' | 'Socratic Coach';
  language: 'English' | 'Hinglish' | 'Hindi';
  studyReminders: boolean;
}
